### PR TITLE
cluster-init: Add `provision aws create-stacks` command

### PR DESCRIPTION
**Pull request by danilo-gemoli**
This PR introduces several things:
- The `ClusterInstall` type. Similarly to the `openshift-installer`'s `install-config.yaml`, it serves as a configuration to drive the cluster installtion process.
- The sub-command: `cluster-init provision aws create-stacks`: provision the required assets on AWS.

The core component here is `pkg/clustermgmt/provision/aws/createstacks.go`.
It leverages [aws-sdk-go-v2](https://github.com/aws/aws-sdk-go-v2) to provision Cloud Formation stacks and performs the following:
- read an AWS configuration
- read a `cluster-install.yaml`
- create any stack specified on the `cluster-install.yaml`, skipping if one exists already
- wait for the stack to be created successfully

### CODE REVIEW

**Review by jmguzik — APPROVED** [review]
I have mostly minor comments and questions but generally LGTM so treat them as optional.

File: pkg/clustermgmt/provision/aws/createstacks_test.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+					{StackName: aws.String("s1"), StackStatus: cloudformationtypes.StackStatusCreateComplete},
+				}}, nil
+			},
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			cfClientGetter := func() (provisionaws.CloudFormationClient, error) {
+				return newFakeCloudFormationClient(tc.onCreateStack, tc.onDescribeStack), nil
+			}
+			wait := 5 * time.Millisecond
```
[the remark this answers is not in the document]
Comment (danilo-gemoli) [reply pkg/clustermgmt/provision/aws/createstacks_test.go:103]: It's an arbitrary small timeout as the default one is just too long.

File: pkg/clustermgmt/provision/aws/createstacks.go [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+			StackName:    &t.StackName,
+			TemplateBody: aws.String(templateBody),
+			Parameters:   params,
+			Capabilities: capabilities,
+		})
+
+		if err != nil {
+			aee := &cloudformationtypes.AlreadyExistsException{}
+			if errors.As(err, &aee) {
+				log.Warn("Stack exists already, skipping")
```
[the remark this answers is not in the document]
Comment (danilo-gemoli) [reply pkg/clustermgmt/provision/aws/createstacks.go:109]: Here is the pattern that I usually follow:
```golang
// Errors go always lowercase
fmt.Errorf("lowercase")

// Log is uppercase
log.WithField("lowercase", "irrelevant here: it depends on the value of this field")
log.Info("Uppercase")
```

File: cmd/cluster-init/cmd/provision.go [hunk 3]
```diff
@@ -40,8 +48,23 @@ func newAWSCreateStacks(ctx context.Context, log *logrus.Entry) *cobra.Command {
...
+			step := aws.NewCreateAWSStacksStep(log,
+				func() (*clustermgmt.ClusterInstall, error) {
+					log.WithField("path", opts.clusterInstall).Info("Loading cluster install")
+					return clustermgmt.LoadClusterInstall(opts.clusterInstall)
+				},
+				func() (aws.CloudFormationClient, error) {
+					log.Info("Loading AWS config")
+					awsconfig, err := config.LoadDefaultConfig(ctx)
+					if err != nil {
+						return nil, fmt.Errorf("load aws config: %v", err)
```
[the remark this answers is not in the document]
Comment (danilo-gemoli) [reply cmd/cluster-init/cmd/provision.go]: That's true but I've noticed that in Go the error message just stated the action that went wrong without specifying that it actually went wrong, by using `failed to`, `couldn't ...`, etc.
```golang
// We are returning an error, therefore the action `load aws config` must be assumed as failed
fmt.Errorf("load was config: %v", err)

// `failed` is redundant IMHO
fmt.Errorf("failed load was config: %v", err)
```

When dumping a chain of errors, this one seems more readable to me:
```golang
create provision aws: load aws config: "/home/dgemoli/aws/.config" does not exist
```
than this one:
```golang
failed to create aws provision: failed to load aws config: "/home/dgemoli/aws/.config" does not exist
```

File: pkg/clustermgmt/provision/aws/createstacks.go [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+	"github.com/openshift/ci-tools/pkg/clustermgmt"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sync/errgroup"
+)
+
+var (
+	createStackCompleteWaitTimeDef time.Duration = 10 * time.Minute
+)
+
+// CloudFormationClient is a conveniente interface that has been created
```
[the remark this answers is not in the document]
Comment (danilo-gemoli) [reply pkg/clustermgmt/provision/aws/createstacks.go]: I know it's verbose but that's what AWS guys suggest: https://aws.github.io/aws-sdk-go-v2/docs/unit-testing/

**Pull request merged**